### PR TITLE
zip+unzip: Port to LibMain :^)

### PR DESCRIPTION
**Pull request by mjz19910**

### CODE REVIEW

File: Userland/Utilities/unzip.cpp [hunk 1]
```diff
@@ -121,17 +112,8 @@ int main(int argc, char** argv)
...
-            perror("mkdir");
-            return 1;
-        }
-
-        rc = chdir(output_directory_path.characters());
-        if (rc < 0) {
-            perror("chdir");
-            return 1;
-        }
+        TRY(Core::System::mkdir(output_directory_path, 0755));
```
Comment (awesomekling) [inline Userland/Utilities/unzip.cpp:115]: This no longer ignores `EEXIST` as before.

**Review by awesomekling — CHANGES_REQUESTED** [review]

**Comment by bgianfo** [comment]
@mjz19910 can you fix the merge conflict?

File: Userland/Utilities/unzip.cpp [hunk 2]
```diff
@@ -83,16 +84,11 @@ int main(int argc, char** argv)
...
 
     String zip_file_path { path };
 
-    struct stat st;
-    int rc = stat(zip_file_path.characters(), &st);
-    if (rc < 0) {
-        perror("stat");
-        return 1;
-    }
+    struct stat st = TRY(Core::System::stat(zip_file_path.characters()));
```
Comment (bgianfo) [inline Userland/Utilities/unzip.cpp:91]: The call to characters() isn't necessary?

Comment (mjz19910) [reply Userland/Utilities/unzip.cpp]: yep, autoconversion will handle that. Is the `.characters()` in the warnln a few lines down necessary?

**Pull request merged**